processor/metrics.go

```diff
@@ -61,17 +62,39 @@ var (
 		stats.UnitDimensionless)
 )
 
+// SpanCountStats represents span count stats grouped by service if DETAILED telemetry level is set,
+// otherwise only overall span count is stored in serviceSpansCounts.
+type SpanCountStats struct {
+	processorName      string
+	serviceSpansCounts map[string]int
+	allSpansCount      int
+	isDetailed         bool
+}
+
+func NewSpanCountStats(td pdata.Traces, processorName string) *SpanCountStats {
+	scm := &SpanCountStats{
+		processorName: processorName,
+		allSpansCount: td.SpanCount(),
+	}
+	if serviceTagsEnabled() {
+		scm.serviceSpansCounts = td.SpanCountByResourceStringAttribute(conventions.AttributeServiceName)
+		scm.isDetailed = true
+	}
```

Comment:
Are you worried that this has a big overhead or why did you try to avoid alway doing this?

Reply:
Yes, I'm worried that it's unnecessary overhead and this feature is disabled most of the times I guess. I assume not many users enable `--metric-level=DETAILED`. Not worrying about it makes the implementation cleaner tho.